### DISCUSSION TITLE
How to attach attributes

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [ben-rogerson/twin.macro#331] [Q&A]
https://styled-components.com/docs/basics#attaching-additional-props

```ts
import { styled } from 'twin.macro'

export const TextInput = styled.input.attrs({ type: 'text' })`p-10`
```

This does not seem to work. No static or runtime errors, yet no effect either (in a Nextjs app context).

**Comment by jasonkuhrt** [comment]
I found this workaround but it doesn't work with intellisense currently:

```ts
import styled from 'styled-components'
import tw from 'twin.macro'

export const TextInput = tw(styled.input.attrs({ type: 'text' })``)`border`

```

**Reply by fvanwijk** [reply]
`import { styled } from 'twin.macro'` does import from styled-components (proxy). So you cannot use border or other twin directives.

This should work:
```
tw.input.attrs({ type: 'text' })`border`
```

**Reply by jasonkuhrt** [reply]
Not quite, at least according to TypeScript:

![image](https://user-images.githubusercontent.com/284476/107660578-6d335500-6c56-11eb-866a-2298112807c5.png)

**Reply by ben-rogerson** [reply]
Sounds like there's a small hole in the types here - PR's welcome 👍

**Reply by jasonkuhrt** [reply]
Ah ok! 

I'll look into that
